### ISSUE TITLE
Soloud missing when building

### ISSUE THREAD

**Issue by EMattfolk** [pythonarcade/arcade#816]
## Bug Report

Trying to build the project with pyinstaller results in the following:

```
Unable to find "c:\users\me\documents\project\venv\lib\site-packages\arcade\soloud\soloud*.dll" when adding binary and data files.
```

The interesting part is that this happens on both windows and linux. I've tried both python3.7 and 3.9.

### Expected behavior:

The program builds.

### Steps to reproduce/example code:

Follow the [tutorial](https://arcade.academy/tutorials/bundling_with_pyinstaller/index.html). It fails using the most simple example.

**Comment by einarf** [comment]
What version? Soloud was removed from from the project in 2.5 in favor of pyglet media/audio.

**Comment by pvcraven** [comment]
My best guess, you aren't using the current version of Arcade. Your environment might be causing the default install to fall back to an earlier Arcade version before we started pinning versions. This can happen with some Python 3.9 setups, but not usually on Windows, so I'm a bit confused on what the issue is.

**Comment by EMattfolk** [comment]
> What version?

I am currently running 2.5.3.

Edit: pyinstaller is version 4.2 if that is any help.

**Comment by pvcraven** [comment]
2.5.3 doesn't use soloud, so I would recheck that version/environment.

**Comment by EMattfolk** [comment]
I have checked multiple times and it seems to be 2.5.3 indeed. The error still persists when using a fresh venv with only arcade and pyinstaller (both latest versions) using python 3.9.1.

From what i can gather there are two lines in __pyinstaller/hook-arcade.py which are highly suspicious.

```python
(hook_path.parent.parent.joinpath("soloud/soloud*.dll"), "./arcade/soloud"), #37
(hook_path.parent.parent.joinpath("soloud/libsoloud.so"), "./arcade/soloud"), #50
```

I will update if I get something working.

**Comment by einarf** [comment]
oh right. I guess the hook needs to be updated. We bundle other libraries now:
https://github.com/pythonarcade/arcade/tree/development/arcade/lib

**Comment by jonas73x** [comment]
I have exactly the same problem, any suggestion on how to fix this? Thx.

**Comment by einarf** [comment]
@SirGnip might have some input here.

**Comment by pvcraven** [comment]
Ok, this is easy to remove, but makes me wonder if we are missing shapely libraries.

**Comment by einarf** [comment]
Yes. shapley and the new bundled libraries are missing in the hook.

**Comment by SirGnip** [comment]
Yeah, if soloud has been removed from the project, it just needs to be removed from `hook-arcade.py` as well.

Now, if there are other dependencies that have been added, (shapely, etc.) they MIGHT or MIGHT NOT need to be explicitly added to `hook-arcade.py`.  Pyinstaller is able to figure a lot of stuff out on its own.  And, some projects provide their own `hook-<PROJ>.py` for Pyinstaller.  So, just adding stuff blindly to `hook-arcade.py` may cause problems.  So, only add dependencies if they are explicitly needed (usually only determined by actually building and running code).

I'd offer to take a look myself, but I won't have time for a few weeks...

**Comment by Cleptomania** [comment]
It should be noted the new sound dependencies are only needed for OS X. Windows and Linux do not have any extra dependencies, not sure how or if Pyinstaller handles that. my guess would be that Pyinstaller would be able to figure out the the new dependencies now as they're implemented a little differently than the soloud ones were, and are loaded automatically through Pyglet. I'm no expert on Pyinstaller though so I could be wrong.

**Comment by SirGnip** [comment]
My understanding is that with Pyinstaller, you just assume that it discovers everything it needs.  Most of the time, this will just happen.  But, during testing, you may hit some kind of dependency that isn't auto-discovered by Pyinstaller.  That is when you know you need to do something explicit (add a hook, etc).  I don't think there is a way to "know" if Pyinstaller will find a dependency (short of reverse engineering Pyinstallers discovery algorithm).

Quote from Pyinstaller docs:  https://pyinstaller.readthedocs.io/en/stable/hooks.html

> The majority of Python packages use normal methods of importing their dependencies, and PyInstaller locates all their files without difficulty. But some packages make unusual uses of the Python import mechanism, or make clever changes to the import system at runtime. For this or other reasons, PyInstaller cannot reliably find all the needed files, or may include too many files. A hook can tell about additional source files or data files to import, or files not to import.

**Comment by Cleptomania** [comment]
I'm gonna close this as the soloud issue was resolved in b4c620e30c745eefc3e4933a185a50a780367f51

If we need to address anything for shapely we can do it in it's own issue when/if it comes up.
